src/platform/amd/acp_7_0/lib/clk.c

```diff
@@ -0,0 +1,492 @@
+
+		clk_rootrefclkmux.bitfields.ROOTREFCLK_MUX_1    = 1;
+		acp_reg_write_via_smn(CLK7_ROOTREFCLK_MUX_1, clk_rootrefclkmux.u32all, sizeof(int));
+
+		clk_spll_fuse1.bitfields.spll_gp_coarse_exp     = 0x5;
+		clk_spll_fuse1.bitfields.spll_gp_coarse_mant    = 0x0;
+		clk_spll_fuse1.bitfields.spll_gi_coarse_exp     = 0x7;
+		clk_spll_fuse1.bitfields.spll_gi_coarse_mant    = 0x0;
+
+		clk_spll_fuse2.bitfields.spll_tdc_resolution    = 0xe8;
+		clk_spll_fuse2.bitfields.spll_freq_offset_exp   = 0xa;
+		clk_spll_fuse2.bitfields.spll_freq_offset_mant  = 0xe;
+
+		clk_spll_field_9.bitfields.spll_dpll_cfg_3      = 2;
+		clk_spll_field_6nm.bitfields.spll_dpll_cfg_4    = 0x60;
+		clk_spll_field_6nm.bitfields.spll_vco_pre_div   = 3;
+		clk_spll_field_7.bitfields.spll_refclk_rate     = 4;
+		clk_spll_field_7.bitfields.spll_pwr_state       = 1;
+		clk_spll_field_4.bitfields.spll_refclk_div      = 0;
+
+		acp_reg_write_via_smn(CLK7_SPLL_FUSE_1, clk_spll_fuse1.u32all, sizeof(int));
+		acp_reg_write_via_smn(CLK7_SPLL_FUSE_2, clk_spll_fuse2.u32all, sizeof(int));
+		acp_reg_write_via_smn(CLK7_SPLL_FIELD_9, clk_spll_field_9.u32all, sizeof(int));
+		acp_reg_write_via_smn(CLK7_SPLL_FIELD_6nm, clk_spll_field_6nm.u32all, sizeof(int));
+		acp_reg_write_via_smn(CLK7_SPLL_FIELD_7, clk_spll_field_7.u32all, sizeof(int));
+		acp_reg_write_via_smn(CLK7_SPLL_FIELD_4, clk_spll_field_4.u32all, sizeof(int));
+
+		clk_spll_field_5nm_bus_wdata.bitfields.bus_spll_wr_data = 0x00400000;
+		acp_reg_write_via_smn(CLK7_SPLL_FIELD_5nm_BUS_WDATA,
+				      clk_spll_field_5nm_bus_wdata.u32all,
+				      sizeof(int));
+
+		clk_spll_field_5nm_bus_ctrl.u32all =
+			acp_reg_read_via_smn(CLK7_SPLL_FIELD_5nm_BUS_CTRL,
+					     sizeof(int));
+		clk_spll_field_5nm_bus_ctrl.bitfields.bus_spll_async_mode   = 1;
+		clk_spll_field_5nm_bus_ctrl.bitfields.bus_spll_apb_mode     = 0;
+		clk_spll_field_5nm_bus_ctrl.bitfields.bus_spll_addr         = 0xa;
+		clk_spll_field_5nm_bus_ctrl.bitfields.bus_spll_byte_en      = 0xf;
+		clk_spll_field_5nm_bus_ctrl.bitfields.bus_spll_rdtr         =
+					!clk_spll_field_5nm_bus_ctrl.bitfields.bus_spll_rdtr;
+		clk_spll_field_5nm_bus_ctrl.bitfields.bus_spll_resetb       = 1;
+		clk_spll_field_5nm_bus_ctrl.bitfields.bus_spll_sel          = 1;
+		clk_spll_field_5nm_bus_ctrl.bitfields.bus_spll_wrtr         = 1;
+		acp_reg_write_via_smn(CLK7_SPLL_FIELD_5nm_BUS_CTRL,
+				      clk_spll_field_5nm_bus_ctrl.u32all,
+				      sizeof(int));
+		do {
+			clk_spll_field_5nm_bus_status.u32all =
+				acp_reg_read_via_smn(CLK7_SPLL_FIELD_5nm_BUS_STATUS,
+						     sizeof(int));
+		} while (clk_spll_field_5nm_bus_status.bitfields.spll_bus_rd_valid !=
+			 clk_spll_field_5nm_bus_ctrl.bitfields.bus_spll_rdtr);
+
+		acp_reg_write_via_smn(CLK7_CLK_PLL_RESET_STOP_TIMER, 0xbbb11aa, sizeof(int));
+		break;
+	default:
+		tr_err(&acp_clk_tr, "ERROR: Invalid PLL Mode");
+		return;
+	}
+
+	clk7_clk_pll_req_u_t  clk7_clk_pll_req;
+
+	clk7_clk_pll_req.u32all = 0;
```

Comment:
`clk7_clk_pll_req` is completely overwritten in the next line, no need to set any members in it